### PR TITLE
Fix: Let Nullable extend Type

### PR DESCRIPTION
**Pull request by localheinz**
This PR

* [x] adds a failing test
* [x] adjusts `Nullable` to extend `Type`

### CODE REVIEW

File: src/Types/NullableType.php [hunk 1]
```diff
@@ -12,8 +12,20 @@
     /** @var Type */
     private $type;
 
-    public function __construct(Nullable $type)
```
Comment (localheinz) [inline src/Types/NullableType.php]: The problem is that we allow injection of implementations of `Nullable` here, but `Nullable` does not extend `Type`:

https://github.com/spatie/typed/blob/2a90dc0d354024867df4028c0a197cf4f7c81945/src/Nullable.php#L9-L12

However, in `NullableType::getType()` we have a return type declaration for `Type`:

https://github.com/spatie/typed/blob/2a90dc0d354024867df4028c0a197cf4f7c81945/src/Types/NullableType.php#L29-L32

**Comment by localheinz** [comment]
https://github.com/spatie/typed/blob/2a90dc0d354024867df4028c0a197cf4f7c81945/src/T.php#L74-L77

should probably also be changed?

**Comment by brendt** [comment]
I was actually thinking about letting the `Nullable` interface extend `Type`. Makes more sense I think? 

That poses one problem though.. Naming 🙄A better name would be `NullableType`, but that one is already used as a real type. They live in a different namespace, but it's my experience that the same names can cause a lot of confusion. Maybe we should call the interface `NullableType`, and call the type `NullType`? 

What's your opinion? It would simultaneously fix the issue this PR addresses because there would be no need for `/** @var Type|Nullable */` and type checking.

**Comment by localheinz** [comment]
@brendt 

> I was actually thinking about letting the `Nullable` interface extend `Type`. Makes more sense I think?

I think this sounds like a good solution!

Would you like me to adjust this PR?

**Comment by brendt** [comment]
That would be awesome! (I only have one more unique star emoji at my disposal though!)

**Comment by brendt** [comment]
Oh and one more remark: we have the convention to write test method names in snake case, no problem if you don't have the time to change it, I can also do that!

**Comment by localheinz** [comment]
@brendt 

Ha, @BackEndTeas has contributed a nice [fixer](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/2.11/src/Fixer/PhpUnit/PhpUnitTestAnnotationFixer.php) for `friendsofphp/php-cs-fixer` for that! Will adjust.

**Comment by brendt** [comment]
Going to merge this tomorrow, as I want to update the README too after merging, and I don't have the time right now.

**Comment by brendt** [comment]
Managed to find the time! I also renamed `NullableType` to `NullType` and used `NullableType` as the interface. Thanks! 🌠

**Comment by localheinz** [comment]
Thank you, @brendt!

**Pull request merged**